clang/lib/AST/ExprConstant.cpp

```diff
@@ -3760,6 +3765,9 @@ findSubobject(EvalInfo &Info, const Expr *E, const CompleteObject &Obj,
     if ((O->isAbsent() && !(handler.AccessKind == AK_Construct && I == N)) ||
         (O->isIndeterminate() &&
          !isValidIndeterminateAccess(handler.AccessKind))) {
+      // Object has ended lifetime since pointer was formed
+      if (handler.AccessKind == AK_IsWithinLifetime)
+        return false;
```

Comment:
AFAIK you can't use `bool` here, it will be problem when `findSubobject` will be used with non-compatible return type

Reply:
There are already other `return false;`/`return true;` returns in `findSubobject`, so `findSubobject::result_type` must be constructible from bool